cache/models/samples.js

```diff
@@ -452,6 +457,309 @@ function upsertOneSample(sampleQueryBodyObj, method, isBulk) {
+    .then((sampleObj) => {
+      if (!sampleObj) {
+        throw new redisErrors.ResourceNotFoundError({
+          explanation: 'Sample not found.',
+        });
+      }
+
+      sampObjToReturn = sampleObj;
+    })
+    .then(() => {
+      cmds.push(redisOps.getHashCmd(aspectType, aspName));
+      cmds.push(redisOps.delKeyFromIndexCmd(sampleType, sampleName));
+      cmds.push(redisOps.delAspFromSubjSetCmd(subjAbsPath, aspName));
+      cmds.push(redisOps.delHashCmd(sampleType, sampleName));
+
+      return redisOps.executeBatchCmds(cmds);
+    })
+    .then((response) => {
+      logObject.dbTime = new Date() - logObject.reqStartTime; // log db time
+      const asp = response[ZERO];
+
+      // attach aspect and links to sample
+      const resSampAsp = cleanAddAspectToSample(sampObjToReturn, asp, method);
+      resSampAsp.isDeleted = new Date().toString();
+      return resSampAsp;
+    });
+  },
+
+  /**
+   * Patch sample. First get sample, if not found, throw error, else get aspect.
+   * Update request body with required fields based on value and related links
+   * if needed. Update sample. Then get updated sample, attach aspect and return
+   * response. Note: Message body and message code will be updated if provided,
+   * else no fields for message body/message code in redis object.
+   * @param  {Object} params - Request parameters
+   * @param  {Object} logObject - Log object
+   * @param  {String} method - Type of request method
+   * @returns {Promise} - Resolves to a sample object
+   */
+  patchSample(params, logObject, method) {
+    const sampleName = params.key.value;
+    const reqBody = params.queryBody.value;
+    let currSampObj;
+    let aspectObj;
+
+    return redisOps.getHashPromise(sampleType, sampleName)
+    .then((sampObj) => {
+      if (!sampObj) {
+        throw new redisErrors.ResourceNotFoundError({
+          explanation: 'Sample not found.',
+        });
+      }
+
+      currSampObj = sampObj;
+      const aspectName = sampleName.split('|')[ONE];
+      return redisOps.getHashPromise(aspectType, aspectName);
+    })
+    .then((aspObj) => {
+      if (!aspObj) {
+        throw new redisErrors.ResourceNotFoundError({
+          explanation: 'Aspect not found.',
+        });
+      }
+
+      cleanQueryBodyObj(reqBody);
+      if (reqBody.value) {
+        aspectObj = sampleStore.arrayStringsToJson(
+          aspObj, constants.fieldsToStringify.aspect
+        );
+
+        const status = sampleUtils.computeStatus(aspObj, reqBody.value);
+        if (currSampObj[sampFields.STATUS] !== status) {
+          reqBody[sampFields.PRVS_STATUS] = currSampObj[sampFields.STATUS];
+          reqBody[sampFields.STS_CHNGED_AT] = new Date().toString();
+          reqBody[sampFields.STATUS] = status;
+        }
+
+        reqBody[sampFields.UPD_AT] = new Date().toString();
+      }
+
+      if (reqBody.relatedLinks) {
+        reqBody[sampFields.RLINKS] = JSON.stringify(reqBody.relatedLinks);
+      }
+
```

Comment:
probably use the fieldsToStringify constant from redis store, even if it is just one field?